tests/Commands/Server/McpServerToolAttributeTests.cs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Reflection;
+using AzureMcp.Commands;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using ModelContextProtocol.Server;
+using NSubstitute;
+using Xunit;
+
+namespace AzureMcp.Tests.Commands.Server;
+
+public class McpServerToolAttributeTests
+{
+    [Fact]
+    public void AllExecuteAsyncMethodsWithMcpServerToolAttribute_ShouldHaveTitleProperty()
```

Comment:
nit - let's combine these attribute tests into a single test that checks for non-null, non-empty, and meaningful.